cds-services-archetype: Changed default JDK version to 25 (#2548)
cap-js/docs: java/migration.md

3. Changed default value of properties:
None

### Changes in the `cds-services-archetype`

The `cds-services-archetype` is used by the `@sap/cds-dk` to generate initial CAP Java projects.

#### Default JDK version

> The default JDK version of new CAP Java projects has been changed to JDK **25**. The minimum required JDK version is still 17, this has not been changed.

Reviewer: ```suggestion
#### Default JDK Version

The default JDK version of new CAP Java projects has been changed to JDK **25**. The minimum required JDK version hasn't changed and is still 17.
```
